src/decode/ykpt/packet_parser/packets.rs

```diff
@@ -0,0 +1,355 @@
+//! Intel PT packets and their constituents.
+
+use deku::prelude::*;
+use std::convert::TryFrom;
+
+/// The `IPBytes` field common to all IP packets.
+///
+/// This tells us what kind of compression was used for a `TargetIP`.
+#[derive(Clone, Copy, Debug, DekuRead)]
+pub(in crate::decode::ykpt) struct IPBytes {
+    #[deku(bits = "3")]
+    val: u8,
+}
+
+impl IPBytes {
+    #[cfg(test)]
+    pub(in crate::decode::ykpt) fn new(val: u8) -> Self {
+        debug_assert!(val >> 3 == 0);
+        Self { val }
+    }
+    /// Returns `true` if we need the previous TIP value to make sense of the new one.
```

Comment:
Missing blank line.

Reply:
I'm sorry, did you say *missing* blank line? I feel like I don't even know you any more :rofl: 

Will fix.

Reply:
As I typed it, I realised that my fingers are not used to that pattern 😆

Reply:
Fixed in [3567abf](https://github.com/ykjit/hwtracer/pull/105/commits/3567abfb722b6106c105a07bc9a0b4ceda47d3a4)